Source/DafnyLanguageServer/Workspace/DocumentManager.cs

```diff
@@ -43,24 +43,24 @@ public class DocumentManager {
     this.documentOptions = documentOptions;
     this.verifierOptions = verifierOptions;
     this.relocator = services.GetRequiredService<IRelocator>();
-    this.notificationPublisher = services.GetRequiredService<INotificationPublisher>();
     this.textChangeProcessor = services.GetRequiredService<ITextChangeProcessor>();
 
     observer = new DocumentObserver(services.GetRequiredService<ILogger<DocumentObserver>>(),
       services.GetRequiredService<ITelemetryPublisher>(),
       services.GetRequiredService<INotificationPublisher>(),
       services.GetRequiredService<ITextDocumentLoader>(),
       document);
-    CompilationManager = new CompilationManager(services,
+    CompilationManager = new CompilationManager(
+      services,
       verifierOptions,
       document,
-      null,
       VerifyOnOpen,
-      ImmutableList.Create<Position>(),
+      ChangedRanges,
       null);
-    observerSubscription = CompilationManager.DocumentUpdates.Subscribe(observer);
+    observerSubscription = CompilationManager.DocumentUpdates.Select(d => d.NotMigratedSnapshot()).Subscribe(observer);
   }
 
+  private const int MaxRememberedChanges = 100;
```

Comment:
Oh maybe I haven't been clear enough. Your new code still keeps looking only at the last 100 changes, but now further restricts to the last 5 verifiables. So if you modify the last verifiable 100 times, you'd get only 1 verifiable at the end, which is what I wanted to avoid. My intention is that you keep the last 5 verifiables no matter the number of changes

In other words, rather than storing and migrating the last 100 changes, you only keep the positions of the last 5 modified verifiables.

Reply:
Are you sure about what the new code does? Notice the `.Distinct()` call in `ChangedVerifiables = freshlyChangedVerifiables.Concat(ChangedVerifiables).Distinct().Take(MaxRememberedChangedVerifiables).ToList()`. I believe this will remember up to 5 _different_ verifiables.

That it also remembers change ranges means that if you make multiple changes quickly, without waiting for translation to occur, it'll still remember those changes and may determine up to 5 last changed verifiables from that.

Reply:
Oh I see. I hadn't seen the concat with the previous ChangedVerifiables. That's correct then !